File: markdowner-core/src/test/java/org/jesperancinha/parser/markdowner/badges/parser/BadgeParser3Test.java
package org.jesperancinha.parser.markdowner.badges.parser;

import org.apache.commons.io.IOUtils;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.charset.Charset;

import static org.assertj.core.api.Assertions.assertThat;

public class BadgeParser3Test {

    @Test
    public void testBadgeParser() throws IOException {
        final var badgeGroups =
                BadgeParser.parse(
                        IOUtils.resourceToString("/badges/Readme3.md", Charset.defaultCharset()));

        assertThat(badgeGroups).isNotNull();
    }

    @Test
    public void testBadgeSettingsParserr() {
        final var badgeSettingGroups =
                BadgeParser.parseSettings();

        assertThat(badgeSettingGroups).hasSize(8);
    }
}

File: markdowner-core/src/test/kotlin/org/jesperancinha/parser/markdowner/badges/parser/BadgeParser3Test.kt
package org.jesperancinha.parser.markdowner.badges.parser

import org.apache.commons.io.IOUtils
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import java.io.IOException
import java.nio.charset.Charset

class BadgeParser3Test {
    @Test
    @Throws(IOException::class)
    fun testBadgeParser() {
        val badgeGroups = BadgeParser.parse(
            IOUtils.resourceToString("/badges/Readme3.md", Charset.defaultCharset())
        )
        Assertions.assertThat(badgeGroups).isNotNull
    }

    @Test
    fun testBadgeSettingsParserr() {
        val badgeSettingGroups = BadgeParser.parseSettings()
        Assertions.assertThat(badgeSettingGroups).hasSize(8)
    }
}